setup.py

```diff
@@ -237,15 +237,46 @@ def check_dependency():
             ['bin/systemd/securetea.service']
         ))
 
+entry_points = {
+    'console_scripts': ['securetea=securetea.entry_points.securetea_core_ep:run_core',
+                        'securetea-server=securetea.entry_points.server_ep:start_server_process [server_req]',
+                        'securetea-system=securetea.entry_points.system_ep:start_system_process [system_req]',
+                        'securetea-iot=securetea.entry_points.iot_ep:start_iot_process [iot_req]']
+}
+
+server_requirements = ["scapy",
+                       "NetfilterQueue",
+                       "pathlib",
+                       "wget",
+                       "yara-python",
+                       "clamd",
+                       "beautifulsoup4",
+                       "lxml",
+                       "clamd"]
+
+system_requirements = ["scapy",
+                       "NetfilterQueue",
+                       "pathlib",
+                       "wget",
+                       "yara-python",
+                       "clamd",
+                       "beautifulsoup4",
+                       "lxml",
+                       "clamd"]
+
+iot_requirements = ["scapy",
+                    "NetfilterQueue",
+                    "shodan"]
+
 setup(
     name='securetea',
-    version='1.5.1',
+    version='2.0',
     packages=find_packages(exclude=["test",
                                     "*.test",
                                     "*.test.*",
                                     "test.*"]),
     data_files=files_definition,
```

Comment:
Do we need data_files? can we include that in manifest.in?

Reply:
We have a logic here going on:
```python
if os_name == 'Ubuntu':
    if int(os_major_version) >= 16:
        files_definition.append((
            '/usr/lib/systemd/system',
            ['bin/systemd/securetea.service']
        ))
if os_name in ['centos', 'redhat', 'debian', 'fedora', 'oracle']:
    files_definition.append((
        '/etc/init.d',
        ['bin/init.d/securetea']
    ))
    if not os_name == 'debian' and int(os_major_version) >= 7:
        files_definition.append((
            '/usr/lib/systemd/system',
            ['bin/systemd/securetea.service']
        ))
```

This logic is for to use SecureTea as a service? Would it be a good idea to hard-code to copy the files directly to the project folder?

Reply:
No. As of now better to remove this. We have to package with deb or rmp later to make this as service. What do you think?

Reply:
We can remove the following parts from file definition now:

```python
if os_name == 'Ubuntu':
    if int(os_major_version) >= 16:
        files_definition.append((
            '/usr/lib/systemd/system',
            ['bin/systemd/securetea.service']
        ))
if os_name in ['centos', 'redhat', 'debian', 'fedora', 'oracle']:
    files_definition.append((
        '/etc/init.d',
        ['bin/init.d/securetea']
    ))
    if not os_name == 'debian' and int(os_major_version) >= 7:
        files_definition.append((
            '/usr/lib/systemd/system',
            ['bin/systemd/securetea.service']
        ))
```

But we do need`data_files` because of the following reasons;
1. `data_files` copies `securetea.conf` in `/etc/securetea`path.

2. `data_files` copies all the internal configuration like internal configuration for Server Log Monitor, System Log Monitor & others in `/etc/securetea` . Also, it copies all the payloads and rules of Server Log Monitor in `/etc/securetea`. This is important because finding a relative path for these text files creates a huge problem & these files need to be present outside the project folder, this will eliminate the problem of finding & generating a relative path.

**For example:**
When running `sudo securetea` inside the home directory terminal will force Python to look for the payloads in the same folder if we use relative path, this will give error. We can't use an absolute path because it's tough to determine python 2.7 & 3.x paths in different OS. The only solution is to use absolute path, here '/etc/securetea/payload_file/....`.

The following snippet is very important:

```python
files_definition = [
    ('/etc/securetea', ['securetea.conf']),
    ('', ['securetea.conf'])
    ('', ['securetea.conf']),
    ('/etc/securetea/asp', ['securetea/lib/auto_server_patcher/configs/commands.json',
                            'securetea/lib/auto_server_patcher/configs/config.json']),
    ('/etc/securetea/log_monitor/server_log/payloads', ['securetea/lib/log_monitor/server_log/rules/payloads/bad_ua.txt',
                                                        'securetea/lib/log_monitor/server_log/rules/payloads/lfi.txt',
                                                        'securetea/lib/log_monitor/server_log/rules/payloads/port_scan_ua.txt',
                                                        'securetea/lib/log_monitor/server_log/rules/payloads/sqli.txt',
                                                        'securetea/lib/log_monitor/server_log/rules/payloads/web_shell.txt',
                                                        'securetea/lib/log_monitor/server_log/rules/payloads/xss.txt']),
    ('/etc/securetea/log_monitor/server_log/regex', ['securetea/lib/log_monitor/server_log/rules/regex/sqli.txt',
                                                     'securetea/lib/log_monitor/server_log/rules/regex/xss.txt']),
    ('/etc/securetea/log_monitor/system_log', ['securetea/lib/log_monitor/system_log/harmful_command.txt']),
    ('/etc/securetea/web_deface', ['securetea/lib/web_deface/config/path_map.json']),
    ('/etc/securetea/antivirus', ['securetea/lib/antivirus/config/config.json'])
]
```

As far as I know, `MANIFEST.in` only copies the file in the project directory, which is not we are looking for. Please share your views.

Reply:
Okay. Keep the data_files for the rest. Remove the code for service implementation

Reply:
@rejahrehim Done :)